File: solutions/fable-1/fable/javascript/wordfreq.js
"use strict";
const input = require("fs").readFileSync(0, "utf8");
const counts = new Map();
const re = /[A-Za-z]+/g;
let m;
while ((m = re.exec(input)) !== null) {
  const w = m[0].toLowerCase();
  counts.set(w, (counts.get(w) || 0) + 1);
}
const entries = [...counts.entries()];
entries.sort((a, b) => (b[1] - a[1]) || (a[0] < b[0] ? -1 : a[0] > b[0] ? 1 : 0));
const out = entries.map(([w, c]) => `${w} ${c}`).join("\n");
process.stdout.write(out ? out + "\n" : "");

File: solutions/fable-1/fable/python/wordfreq.py
import sys
import re
from collections import Counter


def main():
    text = sys.stdin.read()
    words = re.findall(r"[A-Za-z]+", text)
    counts = Counter(w.lower() for w in words)
    items = sorted(counts.items(), key=lambda kv: (-kv[1], kv[0]))
    sys.stdout.write("".join(f"{w} {c}\n" for w, c in items))


if __name__ == "__main__":
    main()
